### ISSUE TITLE
When using custom uniqueValuesStrategy, an error is thrown

### ISSUE THREAD

**Issue by hanastasov** [IgniteUI/igniteui-angular#8461]
When using custom `uniqueColumnValuesStrategy`, an error is thrown

 * igniteui-angular version: 9.1.x

## Steps to reproduce  

1. Open gridCellEditing dev sample and add custom filtering strategy:
```html
    <igx-grid #grid1
        [filterMode]="'excelStyleFilter'"
        [allowFiltering]="true"
        [uniqueColumnValuesStrategy]="columnValuesStrategy"
```
```typescript
    public columnValuesStrategy = (column: IgxColumnComponent,
        columnExprTree: IFilteringExpressionsTree,
        done: (uniqueValues: any[]) => void) => {
        this.getColumnData(column, columnExprTree, uniqueValues => done(uniqueValues));
    }

    public getColumnData(column: IgxColumnComponent,
        columnExprTree: IFilteringExpressionsTree,
        done: (colVals: any[]) => void) {
        setTimeout(() => {
            let columnValues = [];
            if (column.field === 'OrderDate') {
                columnValues = this.gridLocal.data.map(rec => rec[column.field]);
                done(columnValues);
                return;
            }
        }, 1000);
    }
```
3. Open ESF dalog for that column
4. Filter
5. Open again the ESF dialog for that column

## Result  
Error is thrown in the console:
![dateFormat](https://user-images.githubusercontent.com/14248932/97264989-8bec1800-182e-11eb-9cfa-83f55c17292a.gif)

**Comment by DiyanDimitrov** [comment]
@onlyexeption. Could you please check if this issue was resolved after the fix of #8388.

**Comment by onlyexeption** [comment]
@DiyanDimitrov, @hanastasov this issue is resolved after the fix of #8388 !
